src/Bootstrappers/FilesystemTenancyBootstrapper.php

```diff
@@ -106,19 +102,25 @@ public function revert()
         // Storage facade
         Storage::forgetDisk($this->app['config']['tenancy.filesystem.disks']);
         foreach ($this->app['config']['tenancy.filesystem.disks'] as $disk) {
-            /** @var FilesystemAdapter $filesystemDisk */
-            $filesystemDisk = Storage::disk($disk);
-
-            $root = $this->originalPaths['disks']['path'][$disk];
-
-            $filesystemDisk->getAdapter()->setPathPrefix($root);
-            $this->app['config']["filesystems.disks.{$disk}.root"] = $root;
+            $this->updateDiskConfig($disk, 'root', $this->originalPaths['disks']['path'][$disk]);
 
             // Storage Url
-            if ($filesystemDisk->getAdapter() instanceof LocalAdapter && ! is_null($this->originalPaths['disks']['url'])) {
-                $config = $filesystemDisk->getDriver()->getConfig();
-                $config->set('url', $this->originalPaths['disks']['url']);
+            $url = $this->originalPaths['disks.url.' . $disk] ?? null;
+
+            if ($this->getDiskConfig($disk)['driver'] === 'local' && ! is_null($url)) {
```

Comment:
The `getDiskConfig()` method doesn't make much sense. If you're returning null, but using `['driver']` on the return value, that will return in exceptions.

Reply:
You're right, there should be better error handling (would the exception be different from the ones that pieces of [the original code](https://github.com/archtechx/tenancy/commit/eca7b336bc4827b212486a32ce6a2d3df603a918#diff-a7131a9582938c1b642f3d517a9064cbacb7b83d75fb52fe2c6e52dbca26a584R61) would throw?). But if there's no disk with the passed name, the method should return null, that makes sense to me.